### ISSUE TITLE
Block's not working as expected in child elements.

### ISSUE THREAD

**Issue by SeaLife** [twigjs/twig.js#597]
Heyho,

i've been trying to "convert" from PHP to NodeJS and im writing a simple UI using Twig (because im familiar with twig).

// edit:
Using Version 1.12.0
Express Version 4.16.4

My usecase looks like:

```twig
<!-- base.twig -->
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>Deployment API - UI</title>
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta http-equiv="X-UA-Compatible" content="IE=edge"/>
    <link rel="stylesheet" href="https://bootswatch.com/united/../4/united/bootstrap.css" media="screen">
    <link rel="stylesheet" href="https://bootswatch.com/united/../_assets/css/custom.min.css">
</head>
<body>
{% block content %}
{% endblock %}
</body>
</html>
```

```twig
<!-- index.twig -->
{% extends "base.twig" %}

{% block content %}
    <h1>Title Page #1</h1>
{% endblock %}
```

```js
// app.js

const express          = require('express');
const twigFactory      = require('twig');
const app              = express();
const twig             = twigFactory.twig;

app.set("twig options", {
    allow_async     : true, // Allow asynchronous compiling
    strict_variables: false
});

app.route("/").get((req, res) => {
    res.render('index.twig', {
        message: "Test",
        test   : "test"
    });
});
```

im getting the following error:

```bash
Error parsing twig template ***/TraefikAdmin/views/index.twig: 
RangeError: Maximum call stack size exceeded
    at Object.undefined.console.error.e.log.error (***/TraefikAdmin/node_modules/twig/twig.js:1:71530)
    at t (***/TraefikAdmin/node_modules/twig/twig.js:1:69911)
    at ***/TraefikAdmin/node_modules/twig/twig.js:1:78218
    at ***/TraefikAdmin/node_modules/twig/twig.js:1:6598
    at Object.e.attempt (***/TraefikAdmin/node_modules/twig/twig.js:1:71057)
    at e.Thenable.rejectedThen [as then] (***/TraefikAdmin/node_modules/twig/twig.js:1:6572)
    at e.Thenable.catch (***/TraefikAdmin/node_modules/twig/twig.js:1:6339)
    at e.Template.e.parse (***/TraefikAdmin/node_modules/twig/twig.js:1:78202)
    at e.Template.e.parseAsync (***/TraefikAdmin/node_modules/twig/twig.js:1:5510)
    at e.Template.<anonymous> (***/TraefikAdmin/node_modules/twig/twig.js:1:81286)
```

What im expecting?

A rendered Template with the `base.twig` as base and `<h1>Title Page #1</h1>` printed where `{% block contents %}` currently is in `base.twig`.

Regards
SeaLife

**Comment by dave-irvine** [comment]
So it's been a while since I wrote any twig, but doesn't this:

```
<!-- index.twig -->
{% extends "index.twig" %}
```

mean an infinite recursion?

**Comment by ericmorand** [comment]
Yes, definitely, it creates an infinite recursion.

TwigPHP implements a check about that. Your template (without the comment) would throw the following error:

`Circular reference detected for Twig template "index.twig", path: index.twig -> index.twig.`

**Comment by SeaLife** [comment]
Wtf, im stupid while editing. Its actually `{% extends "base.twig" %}`.

Sorry..

**Comment by dave-irvine** [comment]
I think Blocks are fairly broken at the moment, they are being re-written https://github.com/twigjs/twig.js/pull/579

**Comment by toptalo** [comment]
What is your `res.render` code is?

**Comment by willrowe** [comment]
This is now working fine. Fixed by #579.

- [TwigPHP example](https://twigfiddle.com/6gucbz)
- [twig.js example](https://codepen.io/willrowe/pen/wvmrNog)
